File: dev-docs/snippets/database-type-configs/create.swift
import JsBaoClient

// Create a new database type configuration. `databaseType` is required;
// `ruleSetId`, `triggers`, and `metadataAccess` are optional.
func create(client: JsBaoClient, ruleSetId: String) async throws {
  // #region example
  let config = try await client.databaseTypeConfigs.create(
    params: CreateDatabaseTypeConfigParams(
      databaseType: "userDB",
      ruleSetId: ruleSetId,
      triggers: [
        "Task": ["triggers": [["on": "create", "set": ["status": "open"]]]],
      ],
      metadataAccess: "user.role == 'admin'"
    )
  )
  // #endregion example
  _ = config
}

File: dev-docs/snippets/database-type-configs/create.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Create a new database type configuration. `databaseType` is required;
// `ruleSetId`, `triggers`, and `metadataAccess` are optional.
export async function create(client: JsBaoClient, ruleSetId: string) {
  // #region example
  const config = await client.databaseTypeConfigs.create({
    databaseType: "userDB",
    ruleSetId,
    triggers: {
      Task: { triggers: [{ on: "create", set: { status: "open" } }] },
    },
    metadataAccess: "user.role == 'admin'",
  });
  // #endregion example
  return config;
}
